src/plugins/validation/2and3/semantic-validators/refs.js

```diff
@@ -29,21 +25,52 @@ module.exports.validate = function({ jsSpec, specStr, isOAS3 }) {
     match = refRegex.exec(specStr);
   }
 
+  const definitionSectionName = isOAS3 ? 'components' : 'definitions';
   // de-dupe the array, and filter out non-definition refs
   const definitionsRefs = filter(uniq(refs), v =>
-    startsWith(v, `#/${basePath.join('/')}`)
+    startsWith(v, `#/${definitionSectionName}`)
   );
 
-  const definitions = isOAS3 ? jsSpec.components.schemas : jsSpec.definitions;
-  each(definitions, (def, defName) => {
-    if (definitionsRefs.indexOf(`#/${basePath.join('/')}/${defName}`) === -1) {
+  // checks if the definitions are used, and if not, record a warning
+  if (isOAS3) {
+    const definitionTypeList = ['schemas', 'parameters', 'responses'];
+    definitionTypeList.forEach(function(definitionType) {
+      if (jsSpec.components && jsSpec.components[definitionType]) {
+        definitionNotUsedLogger(
+          jsSpec.components[definitionType],
+          definitionsRefs,
+          ['components', definitionType],
+          messages
+        );
+      }
+    });
+  } else {
+    if (jsSpec.definitions) {
+      definitionNotUsedLogger(
+        jsSpec.definitions,
+        definitionsRefs,
+        ['definitions'],
+        messages
+      );
+    }
+  }
+
+  return messages;
+};
+
+function definitionNotUsedLogger(
```

Comment:
I know you like this word, but I'm going to request we don't call this a "logger" - only because at some point I want to add a _real_ logger to this project 🙂 

Maybe use something like `checkIfDefinitionNotUsed`

Reply:
Hahah! Had writer's block initially and meant to change it